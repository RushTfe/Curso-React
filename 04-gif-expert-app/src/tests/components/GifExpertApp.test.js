import React from 'react';
import '@testing-library/jest-dom'

import { shallow } from 'enzyme';
import GifExpertApp from "../../GifExpertApp"

describe('Pruebas sobre el componentes de Gif Expert App', () => {

    test('Debe coincidir con el Snapshot', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect( wrapper ).toMatchSnapshot(); 
    });

    test('Debe de mostrar una lista de categorias', () => {

        const categories = ['OPM', 'DB'];

        const wrapper = shallow( <GifExpertApp defaultCategories={ categories } />);

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    });
});