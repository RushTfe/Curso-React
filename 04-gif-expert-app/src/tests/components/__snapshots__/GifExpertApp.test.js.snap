// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componentes de Gif Expert App Debe coincidir con el Snapshot 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre el componentes de Gif Expert App Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="OPM"
      key="OPM"
    />
    <GifGrid
      category="DB"
      key="DB"
    />
  </ol>
</Fragment>
`;
